proxy-node-gateway/src/main/java/uk/gov/ida/notification/session/GatewaySessionDataValidator.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.ida.notification.session;
+
+import uk.gov.ida.notification.exceptions.SessionAttributeException;
+
+import javax.servlet.http.HttpSession;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.gov.ida.notification.session.SessionKeys.SESSION_KEY_SESSION_DATA;
+
+public class GatewaySessionDataValidator {
```

Comment:
Not sure we need Gateway in the name.

Reply:
It was to be consistent with the name as above.

Reply:
👍